Cannot convert type 'ObjCRuntime.NativeHandle' to 'void*'

### Steps to Reproduce

```cs
IntPtr handle = (IntPtr)1;

// working
void* t1 = (void*)handle;

// error
void* t2 = (void*)(ObjCRuntime.NativeHandle)handle;
```

Related missing method: `IntPTr.ToInt64()`


### Expected Behavior

The code compiles.

### Actual Behavior

```
error CS0030: Cannot convert type 'ObjCRuntime.NativeHandle' to 'void*' 
```

### Environment

<!--
1.
Visual Studio: Help > About Microsoft Visual Studio > Copy Info [button]
Visual Studio for Mac: Visual Studio > About Visual Studio > Show Details > Copy Information [button]
2. Paste into the code block below (between ```)
-->

```
```

### Build Logs

<!--
1. Place cursor below this comment block.
2. Attach build log or link to gist (https://gist.github.com/) of the log.
3. Logs can be found in the Build Output tab of the Errors pad in Visual Studio
To get full build logs, set the log verbosity to diagnostic at the following locations:
- On Visual Studio for Mac: Preferences > Projects > Build
- On Visual Studio for Windows: Tools > Options > Projects and Solutions > Build and Run
On Visual Studio for Windows you also want to add `-v -v -v -v` to the mtouch additional arguments by right-clicking the project in the solution explorer and selecting `Properties`.
-->

### Example Project (If Possible)

<!--
1. Place cursor below this comment block.
2. Drag and drop the compressed project or files needed to reproduce.
-->

<!--
Switch to the "Preview" tab to ensure your issue renders correctly.
-->

--- a/src/ObjCRuntime/NativeHandle.cs
+++ b/src/ObjCRuntime/NativeHandle.cs
@@ -66,6 +66,16 @@ namespace ObjCRuntime {
 			return new NativeHandle (value);
 		}
 
+		public unsafe static explicit operator void* (NativeHandle value)
+		{
+			return (void *) (IntPtr) value;
+		}
+
+		public unsafe static explicit operator NativeHandle (void * value)
+		{
+			return new NativeHandle ((IntPtr) value);
+		}
+
 		public override bool Equals (object? o)
 		{
 			if (o is NativeHandle nh)

--- a/tests/monotouch-test/ObjCRuntime/NativeHandleTest.cs
+++ b/tests/monotouch-test/ObjCRuntime/NativeHandleTest.cs
@@ -1,0 +1,26 @@
+using System;
+
+using Foundation;
+using ObjCRuntime;
+
+using NUnit.Framework;
+
+#if NET
+
+namespace MonoTouchFixtures.ObjCRuntime {
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class NativeHandleTest {
+		[Test]
+		public unsafe void Operators ()
+		{
+			IntPtr value = new IntPtr (0xdadf00d);
+
+			Assert.AreEqual (value, ((NativeHandle) value).Handle, "IntPtr -> NativeHandle");
+			Assert.AreEqual (value, (IntPtr) new NativeHandle (value), "NativeHandle -> IntPtr");
+			Assert.AreEqual (value, ((NativeHandle) ((void *) value)).Handle, "void* -> NativeHandle");
+			Assert.AreEqual (value, (IntPtr) (void *) new NativeHandle (value), "NativeHandle -> void*");
+		}
+	}
+}
+#endif

